submissions/Valerii49/side_menu/js/script.js

```diff
@@ -0,0 +1,151 @@
+		}
+		const activeItem = document.querySelector('._active');
+		if (currentItem !== activeItem) {
+			activeItem.classList.remove('_active');
+			currentItem.classList.add('_active');
+			const itemId = currentItem.dataset.id - 1;
+			buildPage(sitesArr[itemId]);
+		}
+	}
+	else if (currentImage) {
+		const activeImage = document.querySelector('._active-image');
+		if (currentImage !== activeImage) {
+			activeImage.classList.remove('_active-image');
+			currentImage.classList.add('_active-image');
+			const itemIdImage = currentImage.dataset.idImage - 1;
+			const menuItemActivId = document.querySelector('._active').dataset.id - 1;
+			loadHighResImage(sitesArr[menuItemActivId], itemIdImage);
+		}
+	}
+};
+
+function buildPage(data) {
+	const pageMain = document.querySelector('.main-module');
+	let lowResImages = ``;
+	data.gallery.lowRes.forEach((element, index) => {
+		lowResImages +=
+			`<a data-id-image="${index + 1}" class="main-module__image-gallery" href="#">
+			 <img src=${element} alt=${data.gallery.alternateText[index]}>
+		 </a>`});
+	lowResImages = lowResImages.slice(0, 54) + ' _active-image' + lowResImages.slice(54);
+	const mainModuleInner =
+		`<div class="main-module__title">${data.sightName}</div>
+		<div class="main-module__images"><div class="main-module__image"></div><div class="main-module__gallery">${lowResImages}</div></div>
+`;
+	pageMain.innerHTML = mainModuleInner;
+	loadHighResImage(data);
+};
+
+async function loadHighResImage(data, item = 0) {
+	document.querySelector('.main-module__gallery').style.pointerEvents = 'none';
+
+	const styleVar = document.querySelector('.main-module__image').style;
+	styleVar.setProperty('--display-after', 'block');
+
+	try {
+		const url = data.gallery.highRes[item];;
+
+		const blobResult = await getImage(url);
+		const imageUrl = URL.createObjectURL(blobResult);
+
+		const mainImage = document.querySelector('.main-module__image');
+		const mainImageInner = `<img src=${imageUrl} alt="${data.sightName}">
+	<div class="main-module__text"><p>${data.sightAbout[item]}</p></div>`;
+		mainImage.innerHTML = mainImageInner;
+
+		const styleVar = mainImage.style;
+		styleVar.setProperty('--display-after', 'none');
+	} catch (error) {
+		alert(error.message);
+	}
+
+	const styleGalleryVars = document.querySelectorAll('.main-module__image-gallery');
+	styleGalleryVars.forEach((element) => {
+		element.style.setProperty('--display-gallery', 'block');
+	});
+
+	document.querySelector('.main-module__gallery').style.pointerEvents = 'auto';
+
+	async function getImage(url) {
+		const res = await fetch(url);
+		const blob = await res.blob();
+		return blob;
+	};
+}
+
+//*========================================
+buildPage(sitesArr[0]);
+
+function menuBurgerToggle() {
+	document.documentElement.classList.toggle("locked");
+	document.documentElement.classList.toggle("menu-open");
+};
+
+const iconMenu = document.querySelector('.icon-menu');
+iconMenu.addEventListener("click", (e) => {
```

Comment:
You have many listeners that you add, but you don't have removeEventListener don't forget to add it,
if you do not delete them, all your events will be saved in memory, which can lead to memory leak

Reply:
Script has 5 eventListner,  and
iconMenu.addEventListener("click", (e) => {
	document.querySelector('.arrow-main').remove();
}, { once: true });
with option { once: true }
It seems to me that with this option it is deleted automatically.
I did not find where in the script elements with listeners attached to them are removed and added.